src/fireEvent.ts

```diff
@@ -115,10 +115,17 @@ const invokeEvent = (
     return;
   }
 
+  // this is just a placeholder and needs to be a more realistic object
+  const generatedEventObject = { someKey: 'value' };
+
+  let defaultCallbackValues =
+    eventName === 'changeText' ? [] : [generatedEventObject];
+  const handlerCallbackValues = data.length > 0 ? data : defaultCallbackValues;
+
   let returnValue;
 
   act(() => {
-    returnValue = handler(...data);
+    returnValue = handler(...handlerCallbackValues);
   });
```

Comment:
Generally we would like to follow DTL's `fireEvent` implementation, with adjustments necessary to account for differences between RN and React DOM. 

1) They have event map record which holds mapping from event name to event object constructor name and some default init props: https://github.com/testing-library/dom-testing-library/blob/main/src/event-map.js
2) They only apply default event iff someone uses `fireEvent.xxx`. They do not seem to apply in when using `fireEvent('xxx')`: 
- https://github.com/testing-library/dom-testing-library/blob/edffb7c5ec2e4afd7f6bedf842c669ddbfb73297/src/events.js#L5
- https://github.com/testing-library/dom-testing-library/blob/edffb7c5ec2e4afd7f6bedf842c669ddbfb73297/src/events.js#L99
3) Important parts seems to be dispatching that event in context of given node. See: https://github.com/testing-library/dom-testing-library/blob/edffb7c5ec2e4afd7f6bedf842c669ddbfb73297/src/events.js#L17

In DTL they have `element.displatchEvent`, but in ourcase it seems that we have to manually assign `target` property of event, since our `ReactTestInstance` does not implement `EventTarget` which achieves that effect automatically for RTL.